src/catchup/VerifyLedgerChainWork.cpp

```diff
@@ -150,25 +151,35 @@ VerifyLedgerChainWork::verifyHistoryOfSingleCheckpoint()
     {
         if (curr.header.ledgerVersion > Config::CURRENT_LEDGER_PROTOCOL_VERSION)
         {
-            mVerifyLedgerChainFailure.Mark();
             return HistoryManager::VERIFY_STATUS_ERR_BAD_LEDGER_VERSION;
         }
 
         // Verify ledger with local state by comparing to LCL
-        if (curr.header.ledgerSeq == mLastClosed.first + 1)
+        if (curr.header.ledgerSeq == mLastClosed.first)
         {
-            if (curr.header.previousLedgerHash != *mLastClosed.second)
+            if (sha256(xdr::xdr_to_opaque(curr.header)) != *mLastClosed.second)
             {
                 CLOG(ERROR, "History")
                     << "Bad ledger-header history entry: claimed ledger "
                     << LedgerManager::ledgerAbbrev(curr)
                     << " does not agree with LCL "
                     << LedgerManager::ledgerAbbrev(mLastClosed.first,
                                                    *mLastClosed.second);
-                mVerifyLedgerChainFailure.Mark();
                 return HistoryManager::VERIFY_STATUS_ERR_BAD_HASH;
             }
         }
+        // Verify LCL that is just before the first ledger in range
+        else if (curr.header.ledgerSeq == mLastClosed.first + 1 &&
+                 curr.header.ledgerSeq ==
```

Comment:
I don't think you need the second condition: if `curr.header.ledgerSeq == mLastClosed.first + 1` -> the hashes must agree.

Reply:
Added the second condition to avoid potentially checking the same link twice (if LCL is inside the range, it'd be verified later anyway), but it can be removed since the overhead is likely negligible.

Reply:
yeah I agree, it just makes the code harder to read to have this extra condition